airflow/utils/log/file_task_handler.py

```diff
@@ -62,19 +62,29 @@ def __init__(self, base_log_folder: str, filename_template: str | None = None):
                 # handler, not the one that calls super()__init__.
                 stacklevel=(2 if type(self) == FileTaskHandler else 3),
             )
+        self.maintain_propagate: bool = False
 
     def set_context(self, ti: TaskInstance) -> None | SetContextPropagate:
         """
         Provide task_instance context to airflow task handler.
 
+        Generally speaking returns None.  But if attr `maintain_propagate` has
```

Comment:
It might be possible to use mocking instead to avoid having functionality that's used only in unit testing.

Reply:
in this case, to do so, we would have to obliterate code that i'd rather keep. if we mack set_context (or the free function of same name) then we don't execute them.  doing this lets us keep that code live.  it ends up removed in this followup PR (https://github.com/apache/airflow/pull/28571)

which i need to rebase here after some recent changes... and the diff will be horrendous for now don't bother looking :)

Reply:
i've updated it to mark it as `:meta private:` to indicate should not be used by users / not part of public interface / can be removed at any time.

Reply:
I don't really get this one. The `self.maintain_propagate` attribute is available for anyone to check as it is. The point of the method is to set the handler to refer to the provided `ti`.

Reply:
Look at how it is used in the standard task runner test

Reply:
Meta private affects documentation and has been used to signal not public api

Reply:
Not mocking and not replacing entire logging config with a fake one is how I discovered we were creating more connections unnecessarily